### PR TITLE
EES-4368 api data set preview token log

### PR DESCRIPTION
**Pull request by amyb-hiveit**
Adds the preview token log page.

Screenshot 2024-08-05 174424

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/release/data/__tests__/ReleaseApiDataSetPreviewTokenLogPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      },
+    },
+    previousReleaseIds: [],
+  };
+  const testTokens: PreviewToken[] = [
+    {
+      id: 'token-id-1',
+      label: 'Test label 1',
+      status: 'Active',
+      created: new Date('01-01-2024 14:00').toISOString(),
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/data/__tests__/ReleaseApiDataSetPreviewTokenLogPage.test.tsx]: Sorted, don't know why I did that 😆

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenLogPage.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                          <Link
+                            to={generatePath<ReleaseDataSetPreviewTokenRouteParams>(
+                              releaseApiDataSetPreviewTokenRoute.path,
+                              {
+                                publicationId,
+                                releaseId,
+                                dataSetId,
+                                previewTokenId: token.id,
+                              },
+                            )}
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenLogPage.tsx:119]: I've used `useLastLocation` to check the previous page and set the back link appropriately.

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenLogPage.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                            View details
+                            <VisuallyHidden> for {token.label}</VisuallyHidden>
+                          </Link>
+                          <ModalConfirm
+                            title="Revoke this token"
+                            triggerButton={
+                              <ButtonText className="govuk-!-margin-left-2">
+                                Revoke
+                                <VisuallyHidden> {token.label}</VisuallyHidden>
+                              </ButtonText>
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenLogPage.tsx:130]: I've used `useLastLocation` to check the previous page and redirect appropriately after revoking.

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenPage.tsx [hunk 4]
```diff
@@ -109,7 +134,18 @@ export default function ReleaseApiDataSetPreviewTokenPage() {
...
-                  <Link to="todo">View API data set token log</Link>
+                  <Link
+                    to={generatePath<ReleaseDataSetRouteParams>(
+                      releaseApiDataSetPreviewTokenLogRoute.path,
+                      {
+                        publicationId,
+                        releaseId,
+                        dataSetId,
+                      },
+                    )}
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenPage.tsx:145]: Could use the `tokenLogPagePath` variable here?

Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetPreviewTokenPage.tsx]: yep, done now cheers.

**Review by ntsim — APPROVED** [review]

**Pull request merged**